cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -115,6 +131,32 @@ protected MetricsTable getOrCreateMetricsTable(String tableName, DatasetProperti
     return table;
   }
 
+  /**
+   * get the metrics table identified by tableName param if it exists, returns null if it doesn't exist.
+   * @param tableName
+   * @return MetricsTable ; null if the table does not exists
+   * @Nullable
+   */
+  private MetricsTable getResolutionMetricsTable(String tableName) {
+    try {
+      // metrics tables are in the system namespace
+      DatasetId tableId = NamespaceId.SYSTEM.dataset(tableName);
+      MetricsTable table = null;
```

Comment:
I think you can just call getDataset() and it will return null if the table does not exist.